src/validation.cpp

```diff
@@ -681,10 +681,25 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
         }
 
         // do all inputs exist?
-        for (const CTxIn& txin : tx.vin) {
+        for (unsigned int i = 0; i < tx.vin.size(); i++) {
+            const CTxIn& txin = tx.vin[i];
+
             // ELEMENTS:
-            // Don't look for coins that only exist in parent chain
+            // For pegin inputs check whether the pegins have already been claimed before.
```

Comment:
need to note this is just a coin UTXO check, not a mempool check at all. At the mempool level we rely on the `GetConflictTx` check.

Reply:
I added some explanation.